### DISCUSSION TITLE
Développement de modules

### DISCUSSION THREAD

**Discussion by Larsene** [JeromeDevome/GRR#337] [General]
Hello ! Est-ce qu'il y a quelque part un petit guide et les do / don't sur le développement de modules pour GRR ? :-)

**Comment by JeromeDevome** [comment]
Bonjour,
Il n'y a pas de documentation. Et cela est très limité.

Vous pouvez regarder les modules suivants pour comprendre la construction de ceux ci :  https://github.com/JeromeDevome/GRR-Modules

**Comment by Larsene** [comment]
Ah, oui, je vois, assez light ! Bon, c'est pas grave, je vais attaquer la base de données directement, tout y est ! ;-) On va faire une petite solution d'affichage de salles (écrans d'affichages numériques) Ah, oui, je vois, assez light ! Bon, c'est pas grave, je vais attaquer la base de données directement, tout y est ! ;-) On va faire une petite solution d'affichage de salles en interne pour nos écrans d'affichages numériques.

Avec les champs overload ça va le faire. Dommage qu'ils soient spécifiques à un site et pas possiblement à tous !! 😇

Merci !

**Reply by ynaessens** [reply]
Bonjour,
la question a déjà été évoquée : si vous ne faites que lire les informations dans la base de données, il n'y a aucun risque.
Vous accédez à l'affichage sur les écrans en Python ?
Je ne vois pas en quoi les champs additionnels seront utiles, mais j'envisage de rendre les champs transversaux et non plus spécifiques à un domaine (et non site) à la manière des types. Mais cela demande un temps de développement...
Cordialement,
YN

**Reply by Larsene** [reply]
oui, nous utiliserons un programme en Python, effectivement. Les champs additionnels nous permettent d'afficher des informations complémentaires (ex : formateur, réservation, prérequis d'inscriptions, etc). Ces champs sont communs pour nos bâtiments, mais il faut encore pour l'instant en créer autant que nécessaire dans GRR. 

Peut-être un id_area=-1 pour permettre qu'il soit géré pour tous les écrans ?

Evidemment, tout développement nécessite du temps ! ;-)

**Reply by ynaessens** [reply]
Pourquoi ne pas inscrire ces informations dans la description complète de la réservation ?
Pour ce qui est des champs additionnels, historiquement, il était possible de définir des champs de même nom dans des domaines distincts, ce qui mettait le bazar dans certaines pages, notamment de recherche. J'ai donc imposé un couple unique (domaine, champ) pour régler ces problèmes mais cette solution en pose d'autres comme le montre votre demande, et celle d'autres utilisateurs.
Je ne me risquerais pas à forcer $id_area=-1 , mais qui essaiera verra ;-)

**Reply by Larsene** [reply]
Nous n'inscrivons pas ces infos dans la description complète, car nous avons besoin d'afficher ou d'exploiter differemment chaque type d'information. Evidemment, si nos utilisateurs étaient parfaitement rigoureux, nous pourrions imposer une norme pour tout mettre dans ce champ, dans un ordre défini, avec un séparateur reconnu, mais .... Nous n'avons pas ces utilisateurs là ! :-)
D'accord pour les champs additionnels. Après, on fait avec, hein, et ça roule ! ^^

**Comment by ferrierd** [comment]
Bonjour,
Je me permets de rebondir sur cette discussion car je suis moi même à la recherche d'une solution d'affichage du planning sur écran
A ce jour je n'ai pas trouvé de plugin réalisant cette fonctionnalité mais peut-etre ai-je mal cherché.
Si vous aviez des infos/pistes d'une telle fonctionnalité pour afficher un planning GRR sur un écran je suis preneur :-)
Par avance merci
Cdt,
David

**Reply by Larsene** [reply]
Hello @ferrierd , 

Quelques pistes possibles, partant sur ma propre approche :

- Nous avons développé une petite application complémentaire (python / flask) client serveur permettant d'afficher une page web sur un écran, cette page est générée par l'appli selon des templates que nous avons développés également (cette application peut gérer plusieurs dizaines d'écrans avec une personnalisation par écran)
- Chaque écran est géré par un Raspberry avec [FullPageOs ](https://github.com/guysoft/FullPageOS)(un système d'exploitation qui ne permet que l'affichage d'une page web), auquel on greffe une appli python (gestion des redémarrage, extinction de l'écran via DPMS (pas encore au top)
- Il existe également [Yodeck ](https://www.yodeck.com)(mais payant), qui peut te permettre de gérer tes écrans de façon plus complète, et de gérer des pages web en affichage (mais pas seulement : des vidéos, photos, pdf, etc - ce qui permet de faire tourner l'affichage de salles avec du contenu multimédia) 
- Attaque directement la base de données de GRR, elle est très simple à comprendre. quelques requêtes sur les grr_entry, qrr_area, grr_room peuvent suffire
- pour afficher précisément certains éléments complémentaires, tu peux utiliser le champ "overload_desc" dont le contenu est décrit dans la table grr_overload. (la lecture se fait via https://github.com/JeromeDevome/GRR/blob/6e529d1b3dde87621f0da83073e51c42109ac01d/include/mrbs_sql.inc.php#L386)

**Reply by ferrierd** [reply]
Merci bcp @Larsene pour ce retour complet.
Cela me donne de bonnes pistes d'étude
Cdt,
